adele-order-manager/src/main/java/com/microservicesteam/adele/ordermanager/domain/OrderRepository.java

```diff
@@ -1,14 +1,31 @@
 package com.microservicesteam.adele.ordermanager.domain;
 
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface OrderRepository extends JpaRepository<Order, String> {
 
-    @Query("update Order o set o.paymentId = :paymentId where orderId = :orderId")
-    void updatePaymentId(@Param("orderId")String orderId, @Param("paymentId") String paymentID);
+    @Modifying(clearAutomatically = true)
```

Comment:
Our investigation revealed, the following:
1. `@Modifying` annotation is necessary, because all repository "methods" are queries by default (as the `@Query` annotation suggests), therefore this annotation is needed if the query is not a select by a DML command. See [Modifying Queries](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.modifying-queries)
2. It seems that the `clearAutomatically` parameter is needed only to make the tests "work", since Hibernate cannot register changes outside of the persistent context, like here as the payment id is updated. This parameter forces the entity manager to clear it's context once the statement is executed, therefore all entities needs to be related by Hibernate again. Unfortunately there are downsides of either setting this to true or false. I suggests to discuss this offline.

Reply:
@bearcheese entityManager.clear() is our friend. Must be called after each update. We don't need clearAutomatically in production code then.